internal/docker/docker_client_test.go

```diff
@@ -0,0 +1,91 @@
+package docker
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/client"
+)
+
+// Write a test that asserts that we are not overwriting the DOCKER_HOST environment variable
+
+func Test_newClient(t *testing.T) {
+	cases := []struct {
+		name           string
+		providedSocket string
+		expectedSocket string
+		setEnv         func() func()
+	}{
+		{
+			name:           "Test newClient returns the correct default location",
+			providedSocket: "",
+			expectedSocket: "unix:///var/run/docker.sock",
+		},
+		{
+			name:           "Test newClient with runtime specific path",
+			providedSocket: "",
+			setEnv: func() func() {
+				os.Setenv("DOCKER_HOST", "unix:///var/CUSTOM/docker.sock")
+				return func() {
+					os.Unsetenv("DOCKER_HOST")
+				}
```

Comment:
```suggestion
			setEnv: func(t *testing.T) func() {
				t.Setenv("DOCKER_HOST", "unix:///var/CUSTOM/docker.sock")
```

- testing will automatically re-set the value back to what it was before the test
- pass `t *testing.T` explicitly so that it is scoped to the right test at cleanup